### DISCUSSION TITLE
Upgrading keycloak to any version beyond 20.0.5 gives issues with admin console login

### DISCUSSION THREAD

**Discussion by spoorthyredi2** [keycloak/keycloak#22984] [Archive]
Hi Team , 

We are using v20.0.1 with some custom themes , for a while now.  We are looking to upgrade to v21.
However, when we upgrade to any minor version of v21 or v22, (tried v21.1.1, v21.1.2 , v21.0.1 , v21.0.2 , v22.0.0, v22.0.1 )  admin console login fails with an internal server error.   It works with base keycloak theme, but not with any custom theme. We need to use custom themes as we cannot go back to old pages . Please let us know what can be done here ... 

We get below exception : 
```

        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: freemarker.template.TemplateNotFoundException: **Template not found for name "index.ftl".**
The name was interpreted by this TemplateLoader: org.keycloak.theme.freemarker.DefaultFreeMarkerProvider$ThemeTemplateLoader@48c6cb2.
        at freemarker.template.Configuration.getTemplate(Configuration.java:2883)
        at freemarker.template.Configuration.getTemplate(Configuration.java:2703)
        at org.keycloak.theme.freemarker.DefaultFreeMarkerProvider.getTemplate(DefaultFreeMarkerProvider.java:66)
        at org.keycloak.theme.freemarker.DefaultFreeMarkerProvider.processTemplate(DefaultFreeMarkerProvider.java:39)
        ... 47 more

```

**Comment by jonkoops** [comment]
Your theme probably tries to extend the old Admin Console theme, which no longer exist. If you want to continue to use a custom theme, you will have to update it to match the structure of the new Admin Console theme.

**Reply by spoorthyredi2** [reply]
@jonkoops  Thanks for the way ahead, Is there a specific document on the structure of v2 admin theme that you can point us to? 


Our current directory structure for admin theme inside a custom theme is - 
themes/<Customtheme folder>/admin/resources/img
themes/<Customtheme folder>/admin/resources/css
themes/<Customtheme folder>/admin/theme.properties


Contents of theme.properties

```
parent=base
import=common/keycloak

styles=css/styles.css
stylesCommon=node_modules/patternfly/dist/css/patternfly.min.css node_modules/patternfly/dist/css/patternfly-additions.min.css node_modules/select2/select2.css lib/angular/treeview/css/angular.treeview.css node_modules/text-security/text-security.css


# This is the logo in upper lefthand corner.
# It must be a relative path.
logo=/img/leaplogo.png

# This is the link followed when clicking on the logo.
# It can be any valid URL, including an external site.
logoUrl=./

# This is the icon for the account console.
# It must be a relative path.
favIcon=/img/favicon.ico
```

**Reply by jonkoops** [reply]
We're working on updated documentation for the customization of the Admin Console, for more information see #19508.

**Reply by spoorthyredi2** [reply]
We have a custom theme called 'leap' under which we have 5 sub themes .  login/account/admin/welcome/common . 
Have tried to extend v2 by using  _parent=keycloak.v2_ instead of _parent=base_.  in all those sub themes.

Now I get this error 
 **Not found parent theme 'keycloak.v2' of theme 'leap'. Unable to load LOGIN theme 'leap' due to this.**

I'm now running v22.0.0 and I can login to admin console. But the theme on UI are all default keycloak. Not the custom theme.
@jonkoops can you help us here.. 

Our folder structure
<img width="314" alt="image" src="https://github.com/keycloak/keycloak/assets/116352972/6e9597b5-1420-446f-b57a-ecaeb82a4f33">

**Reply by jonkoops** [reply]
If I had to guess, we only have a `.v2` for the Admin Console. You can't extend it for any other theme.

**Reply by spoorthyredi2** [reply]
I can see v2 for account given at https://github.com/keycloak/keycloak/tree/main/themes/src/main/resources/theme/keycloak.v2 . And it worked for me .. Login page showed Custom theme and logged in successfully . But post login. the ADMIN UI is still on DEFAULT theme. What should be the structure for Admin v2 ? @jonkoops  @abbra

**Reply by jonkoops** [reply]
You can take a look at how we're [building](https://github.com/keycloak/keycloak/blob/main/js/apps/admin-ui/pom.xml) the new Admin Console. For now the documentation is lacking, so I have no further information unfortunately.
